### PR TITLE
Added plugin octo_encoder_filament

### PR DESCRIPTION
**Pull request by Akex2**

### CODE REVIEW

File: _plugins/octo_encoder.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+layout: plugin
+
+id: octo_encoder
+title: Octo_Filament_Encoder
+description: filament checker with encoder
+author: Alex
+license: AGPLv3
+
+# TODO
+date: 2018-05-10
```
Comment (kantlivelong) [inline _plugins/octo_encoder.md]: Can you adjust this to the proper initial release date(date of PR or release on GitHub)?

**Comment by kantlivelong** [comment]
Hi @Akex2,

Thanks for the submission. I've taken a quick look at this PR and the source for the plugin and have a few comments/questions/concerns:

- As noted, can you fix the date?
- I noticed you are missing a [software update hook](http://docs.octoprint.org/en/master/bundledplugins/softwareupdate.html). I would highly encourage doing this before I merge this PR. Otherwise users will not be able to receive updates.
- It looks like you didn't change the [name of the class ](https://github.com/Akex2/Octo_Filament_Encoder/blob/647d8b99d1e7759ee539d0a9880de9de987350f4/octo_filament_encoder/__init__.py#L71 )which is from the sample. Could you give it a more unique name? Preferably one that identifies with the plugin.
- I noticed you have [two classes ]
(https://github.com/Akex2/Octo_Filament_Encoder/blob/647d8b99d1e7759ee539d0a9880de9de987350f4/octo_filament_encoder/__init__.py#L26-L54 )defined outside of the plugin class. While not required, I would recommend moving them within the the plugin class.

**Comment by foosel** [comment]
Hey @Akex2! First of all huge apologies for letting this slip through the cracks. I somehow never saw your update and thought this was still outstanding.

I wanted to ask if after all this time you are even still interested in getting this merged? If so I'll gladly take a final look.

**Comment by Akex2** [comment]
Hi foosel :)
yes of course please !

**Comment by foosel** [comment]
Hey, took another look:

  * Your plugin states that I'm and the OctoPrint Project are the author, I'm not okay with that ;)

    https://github.com/Akex2/Octo_Filament_Encoder/blob/master/octo_filament_encoder/__init__.py#L4-L6

  * You've checked in a pyc file, that should be removed (it's a compiled version of your code that might cause issues on target systems)

    https://github.com/Akex2/Octo_Filament_Encoder/blob/master/octo_filament_encoder/__init__.pyc

  * Your plugin still doesn't register itself with the update mechanism, as already mentioned by @kantlivelong. You really should change that, otherwise it'll make it really difficult for you to properly maintain your plugin for your users. You can find information on that [here](https://plugins.octoprint.org/help/registering/).

  * Your plugin's README is incomplete.

    https://github.com/Akex2/Octo_Filament_Encoder/blob/master/README.md

  * Finally please be aware that newly registered plugins from now one should also be tested and made sure to be compatible to Python 3 (with OctoPrint 1.4.0rc3), since Python 2 is EOL. You can find more on that [here](https://plugins.octoprint.org/help/registering/) and [here](https://community.octoprint.org/t/towards-python-3-and-octoprint-1-4-0/12382). Ideally take care of this as well now please, otherwise you'll have to do it at a later date or your plugin at some point will become obsolete.

Please take care of these points and ping me here once done.

**Comment by jneilliii** [comment]
@Akex2 any updates as to the status of the above mentioned required changes?

**Comment by Akex2** [comment]
I will keep it just for me
I will closed this PR
Thx

**Pull request closed without merging**